extensions/hibernate-orm/runtime/src/main/java/io/quarkus/hibernate/orm/runtime/cdi/QuarkusManagedBeanRegistry.java

```diff
@@ -0,0 +1,91 @@
+package io.quarkus.hibernate.orm.runtime.cdi;
+
+import org.hibernate.resource.beans.container.spi.BeanContainer;
+import org.hibernate.resource.beans.container.spi.ContainedBean;
+import org.hibernate.resource.beans.internal.FallbackBeanInstanceProducer;
+import org.hibernate.resource.beans.spi.ManagedBean;
+import org.hibernate.resource.beans.spi.ManagedBeanRegistry;
+
+import io.quarkus.arc.Arc;
+
+/**
+ * A replacement for ManagedBeanRegistryImpl that:
+ * <ul>
+ * <li>forces the use of QuarkusManagedBeanRegistry,
+ * which works with Arc and respects configured scopes when instantiating CDI beans.</li>
+ * <li>is not stoppable and leaves the release of beans to {@link QuarkusArcBeanContainer},
+ * so that the bean container and its beans can be reused between static init and runtime init,
+ * even though we stop Hibernate ORM services after the end of static init.</li>
```

Comment:
Ok, I'm not entirely sure this will be safe enough and if, at some point, we will have to release the beans and then reinitialize them. Let's get this in and we will see if it ends up being an issue.